provider/docker_test.go

```diff
@@ -149,12 +157,15 @@ func TestDockerGetFrontendRule(t *testing.T) {
 		},
 	}
 
-	for _, e := range containers {
-		dockerData := parseContainer(e.container)
-		actual := provider.getFrontendRule(dockerData)
-		if actual != e.expected {
-			t.Fatalf("expected %q, got %q", e.expected, actual)
-		}
+	for containerID, e := range containers {
+		t.Run(strconv.Itoa(containerID), func(t *testing.T) {
```

Comment:
You need to make an assignment like `e := e` before calling `t.Run()` in order to capture the range variable. Otherwise, your variable won't get bound to the correct instance.

It's basically about [this particular Go gotcha](https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables).